src/components/map/rent-popup.tsx

```diff
@@ -87,6 +87,23 @@ const RentPopup: React.FC<RentPopupProps> = ({
       });
   }, [booking, fetchBooking, fetchStationDetail, selectedStation, BUCHUNGEN]);
 
+  const handleRefreshBooking = useCallback(() => {
+    if (!booking) {
+      throw new Error('Trying to refresh a booking, but no bike booked.');
+    }
+    if (!selectedStation) {
+      throw new Error('Trying to refresh a booking, but no station selected.');
+    }
+
+    return cancelBooking()
+      .then(() => bookBike(selectedStation.stationId))
+      .then(() => Promise.all([fetchBooking(), fetchStationDetail()]))
+      .catch(err => {
+        console.error('Error while refreshing a booking:', err);
+        toast(BUCHUNGEN.ALERT.LOAD_CURR_BOOKING_ERR, { type: 'error' });
+      });
+  }, [booking, fetchBooking, fetchStationDetail, selectedStation]);
+
```

Comment:
> During the canceling and rebooking the UI will plop nevertheless if we don't (there is a moment in between where booking is falsy)

I overlooked this during the first review.

This must be done differently. Booking refresh mustn't cause "weird" intermediate state updates. The hook should expose a function does the refreshing and only when that is done (or has errored) the hook should update its public state (and thus update the UI).

Reply:
527138892141840f66cff9dfb6c672aa26edb739I commited the current state regarding this problem. I wrote a new callback that should handle canceling and rebooking without setting `booking` to `null`. However, it only cancels the booking at the moment.

Reply:
As far as I see this, the callback inside the view can be removed entirely in favor of just calling the hook function directly as event handler.

Reply:
The callback should not go away, because we still need to refresh the station detail when we're done refreshing the booking. We just want to remove `fetchBooking` from here (because the hook is supposed to spit out a correct booking at any time).